src/yarn-linter.js

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env node
+
+const fs = require('fs');
+
+class YarnLinter {
+
+  constructor(context) {
+    this.fsOptions = {
+      encoding: 'utf-8'
+    };
+
+    if (this.validateContext(context)) {
+      this.validateYarnInstall(context);
+    }
+  }
+
+  validateContext(context  = null) {
```

Comment:
with two "validator" functions so far, maybe there's a case to make some sort of `YarnLinterValidationService` or something and DI it in?